Code/Tools/ProjectManager/tests/GemCatalogTests.cpp

```diff
@@ -35,30 +43,530 @@ namespace O3DE::ProjectManager
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[CityProps], QModelIndex()));
+    }
+
+    TEST_F(GemCatalogMiscFilterTests, GemCatalog_Filters_By_Type_Only)
+    {
+        m_proxyModel->SetTypes(GemInfo::Type::Code);
+
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[DefaultAudio], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[MobileUX], QModelIndex()));
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[CityProps], QModelIndex()));
+
+        m_proxyModel->SetTypes(GemInfo::Type::Tool);
+
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[DefaultAudio], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[MobileUX], QModelIndex()));
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[CityProps], QModelIndex()));
+
+        m_proxyModel->SetTypes(GemInfo::Type::Asset);
+
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[DefaultAudio], QModelIndex()));
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[MobileUX], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[CityProps], QModelIndex()));
+    }
+
+    TEST_F(GemCatalogMiscFilterTests, GemCatalog_Filters_By_Type_Multiselection_Shows_All_Matches)
+    {
+        m_proxyModel->SetTypes(GemInfo::Type::Tool | GemInfo::Type::Asset);
+
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[DefaultAudio], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[MobileUX], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[CityProps], QModelIndex()));
+    }
+
+    TEST_F(GemCatalogMiscFilterTests, GemCatalog_Filters_By_Platform_Only)
+    {
+        m_proxyModel->SetPlatforms(GemInfo::Platform::Windows);
+
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[DefaultAudio], QModelIndex()));
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[MobileUX], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[CityProps], QModelIndex()));
+
+        m_proxyModel->SetPlatforms(GemInfo::Platform::Android);
+
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[DefaultAudio], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[MobileUX], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[CityProps], QModelIndex()));
+
+        m_proxyModel->SetPlatforms(GemInfo::Platform::macOS);
+
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[DefaultAudio], QModelIndex()));
+        EXPECT_FALSE(m_proxyModel->filterAcceptsRow(m_gemRows[MobileUX], QModelIndex()));
+        EXPECT_TRUE(m_proxyModel->filterAcceptsRow(m_gemRows[CityProps], QModelIndex()));
+    }
+
+    TEST_F(GemCatalogMiscFilterTests, GemCatalog_Filters_By_Platform_Multiselection_Shows_All_Matches)
+    {
+        m_proxyModel->SetPlatforms(GemInfo::Platform::Windows | GemInfo::Platform::iOS);
```

Comment:
This sets up a different filtering scenario than the other multiselects, in that all three elements are found. The negative filtering case isn't being verified, due to how the platform labels line up.

I don't see a good filter that causes two different groups singly match, so perhaps modify the registered platform info to make sure one row gets excluded by the filter?